### PR TITLE
state_vectors to state_kets

### LINKED ISSUE #29: Make AtomicEnsemble.state_vectors into Kets
As discussed here: https://github.com/bleykauf/aisim/pull/23#discussion_r436255687

### PR DESCRIPTION
**Pull request by savowe**
If finished, it closes #29 #
Next step is to adjust the transition method for new dimensions of state_vectors -> state_kets

### CODE REVIEW

File: aisim/atoms.py [hunk 1]
```diff
@@ -35,10 +35,13 @@ class AtomicEnsemble():
...
 
-    def __init__(self, phase_space_vectors, state_vectors=[1, 0], time=0, weights=None):
+    def __init__(self, phase_space_vectors, state_kets=[1, 0], time=0, weights=None):
         assert phase_space_vectors.shape[1] == 6
         self.phase_space_vectors = phase_space_vectors
-        self.state_vectors = state_vectors
+        if isinstance(state_kets, list):
+            self.state_kets = np.array([state_kets]).T
+        else:
+            self.state_kets = state_kets
```
Comment (bleykauf) [inline aisim/atoms.py:44]: This is all alrady done by the setter of the state_ket property!

File: aisim/atoms.py [hunk 2]
```diff
@@ -70,22 +73,33 @@ def time(self, value):
...
-    def state_vectors(self, new_state_vectors):
-        new_state_vectors = np.array(new_state_vectors)
-        if new_state_vectors.ndim == 1:
+    @state_kets.setter
+    def state_kets(self, new_state_kets):
+        if isinstance(new_state_kets, list):
+            new_state_kets = np.array([new_state_kets]).T
+        else:
+            new_state_kets = new_state_kets
+        if len(new_state_kets) == 2:
```
Comment (bleykauf) [inline aisim/atoms.py:88]: what was wrong with the old test, where the ndim attribute is checked? Now it only works for 2 level systems

File: aisim/atoms.py [hunk 3]
```diff
@@ -70,22 +73,33 @@ def time(self, value):
...
+        return self._state_kets
 
-    @state_vectors.setter
-    def state_vectors(self, new_state_vectors):
-        new_state_vectors = np.array(new_state_vectors)
-        if new_state_vectors.ndim == 1:
+    @state_kets.setter
+    def state_kets(self, new_state_kets):
+        if isinstance(new_state_kets, list):
+            new_state_kets = np.array([new_state_kets]).T
```
Comment (bleykauf) [inline aisim/atoms.py:85]: same thing as in __init__: the transpose does nothing

File: aisim/atoms.py [hunk 4]
```diff
@@ -70,22 +73,33 @@ def time(self, value):
...
+        else:
+            new_state_kets = new_state_kets
```
Comment (bleykauf) [inline aisim/atoms.py:87]: again, this does nothing

File: aisim/atoms.py [hunk 5]
```diff
@@ -141,7 +155,7 @@ def calc_position(self, t):
         return self.initial_position + self.initial_velocity * t
 
     def state_occupation(self, state):
-        return np.abs(self.state_vectors[:, state])**2
+        return np.real(np.einsum('ijk,ijk->ij' , self.state_bras, self.state_kets)[:,state])
```
Comment (bleykauf) [inline aisim/atoms.py:158]: trailing whitespace after string

will try to setup automatic checking with flake8 in PRs soon...

File: aisim/atoms.py (hunk 1 above)
Comment (savowe) [reply aisim/atoms.py]: yes. removed it.

File: aisim/atoms.py (hunk 3 above)
Comment (savowe) [reply aisim/atoms.py:82]: thats not true.
if you want to set the state_ket with the list new_state_vectors = [1,0], then the np.array([new_state_kets]) part maks it a numpy array of shape (2,1) and the transpose makes it an actual row vector

File: aisim/atoms.py (hunk 4 above)
Comment (savowe) [reply aisim/atoms.py]: yes. I removed the else statement

File: aisim/atoms.py (hunk 2 above)
Comment (savowe) [reply aisim/atoms.py]: yes this was not good. changed it again to check for two dimensions (row and columns of state_vector).
complete sets of kets should have three dimensions

File: aisim/atoms.py (hunk 3 above)
Comment (savowe) [reply aisim/atoms.py:82]: sorry, it is first (1,2) and transpose makes it (2,1)

Comment (bleykauf) [reply aisim/atoms.py:82]: true, didn't see the additional list in `[new_state_kets]`

**Pull request merged**